### PR TITLE
stringify http expression before passing to expression render

### PR DESCRIPTION
**Pull request by tianleh**
*Issue #, if available:*

*Description of changes:*
Stringify input to the expression modal to fix the broken display in current authc & authz page.

*Test*
Current error
Screen Shot 2020-08-18 at 5 53 53 PM

After the fix
Screen Shot 2020-08-18 at 5 54 16 PM

Also verify that the fix didn't impact the role view page.
Screen Shot 2020-08-18 at 5 54 33 PM

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by yuxitang-amzn** [comment]
Is object always able to stringify? If yes, I would suggest doing the opposite way (pass object as parameter). So we don't need to worry about parsing failure inside the component.

**Comment by tianleh** [comment]
> Is object always able to stringify? If yes, I would suggest doing the opposite way (pass object as parameter). So we don't need to worry about parsing failure inside the component.

Per https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify , it should be able to stringify. I wrote it in such a way to avoid leaving a generic "any" type for such object by passing string between the functions.

**Comment by tianleh** [comment]
> > Is object always able to stringify? If yes, I would suggest doing the opposite way (pass object as parameter). So we don't need to worry about parsing failure inside the component.
> 
> Per https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify , it should be able to stringify. I wrote it in such a way to avoid leaving a generic "any" type for such object by passing string between the functions.

Decided to switch to pass object since string can be arbitrary

File: public/apps/configuration/panels/role-view/index-permission-panel.tsx [hunk 1]
```diff
@@ -139,8 +139,8 @@ export function IndexPermissionPanel(props: IndexPermissionPanelProps) {
   return (
     <PanelWithHeader
       headerText={headerText}
-      headerSubText="Index permissions allow you to specify how users in this role can access the indices. You can restrict a role to a subset of documents in an index, 
-      and which document fields a user can see as well. If you use field-level security in conjunction with document-level security, 
+      headerSubText="Index permissions allow you to specify how users in this role can access the indices. You can restrict a role to a subset of documents in an index,
+      and which document fields a user can see as well. If you use field-level security in conjunction with document-level security,
```
Comment (yuxitang-amzn) [inline public/apps/configuration/panels/role-view/index-permission-panel.tsx:143]: Will this change affect the text?

Comment (tianleh) [reply public/apps/configuration/panels/role-view/index-permission-panel.tsx:149]: It is a ts warning that there is a trailing empty space. Just fixed it along the way.

**Pull request merged**